src/test/java/com/mparticle/kits/ButtonKitTests.java

```diff
@@ -139,4 +179,50 @@ private static void setFinalStatic(Field field, Object newValue) throws Exceptio
 
         field.set(null, newValue);
     }
-}
\ No newline at end of file
+
+    private class TestKitManager extends KitManagerImpl {
+        private Map<String, String> attributes = new HashMap<>();
+        private AttributionResult result;
+        private AttributionError error;
+
+        TestKitManager() {
+            super(context, null, null, null, null);
+        }
+
+        @Override
+        Map<String, String> getIntegrationAttributes(KitIntegration kitIntegration) {
+            return attributes;
+        }
+
+        @Override
+        void setIntegrationAttributes(KitIntegration kitIntegration,
+                Map<String, String> integrationAttributes) {
+            this.attributes = integrationAttributes;
+        }
+
+        @Override
+        public void onResult(AttributionResult result) {
+            this.result = result;
+        }
+
+        @Override
+        public void onError(AttributionError error) {
+            this.error = error;
+        }
+    }
+
+    private class TestKitConfiguration extends KitConfiguration {
+        @Override
+        public int getKitId() {
+            return TEST_KIT_ID;
+        }
+    }
+
+    private class TestMParticle extends MParticle {
+        @NonNull
+        @Override
+        public IdentityApi Identity() {
```

Comment:
I guess this is why we should try not to have static methods/singletons in the constructor 😬